lib/spack/spack/package.py

```diff
@@ -1464,7 +1467,27 @@ def do_install(self, **kwargs):
 
     @property
     def test_log_name(self):
-        return 'test-%s-out.txt' % self.spec.format('{name}-{version}-{hash:7}')
+        return 'test-{0}-out.txt' \
+            .format(self.spec.format('{name}-{version}-{hash:7}'))
+
+    @property
+    def test_dir(self):
+        return self.test_stage.join(
+            self.spec.format('{name}-{version}-{hash}'))
+
+    def copy_src_to_install(self, src_subdir, install_subdir):
+        """Copy the source subdirectory to the install test subdirectory.
+
+        Args:
+            src_subdir (str): name of the subdirectory under staged source
```

Comment:
Can you make this accept either a file or a directory? I think some packages will have way more test infrastructure than we want to take.

Reply:
Accepts a list of relative paths.